test/server/controller.spec.js

```diff
@@ -712,360 +814,599 @@ describe('Redirection in json response', function() {
+      let obj = JSON.parse(JSON.stringify(optionsList));
+      if (obj.method === "GET") {
+        obj.path = server_path + '?referenceName=chr1&start=400&end=4';
+      } else {
+        obj.path = server_path;
+      }
+      let req = http.request( obj, function(response) {
+        var body = '';
+        response.on('data', function(d) { body += d;});
+        response.on('end', function() {
+          checkGenericErrorResponse(expect, response);
+          expect(response.statusCode).toEqual(400);
+          let errorPayload = JSON.parse(body);
+          expect(errorPayload.htsget).toBeDefined();
+          expect(errorPayload.htsget.error).toBe(ServerHttpError.INVALID_INPUT);
+          expect(response.statusMessage).toEqual(
+            'Range end should be bigger than start');
+          done();
+        });
+      });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam", "accession":id, "regions":[{"referenceName":"chr1", "start":400, "end":4}]}), () => {
+          req.end();
+        });
+      } else {
+        req.end();
+      }
     });
-  });
 
-  it('redirection error, invalid characted in reference name', function(done) {
-    http.get(
-      { socketPath: socket,
-        path: server_path + '?referenceName=chr@1&start=400&end=4'}, function(response) {
-      var body = '';
-      response.on('data', function(d) { body += d;});
-      response.on('end', function() {
-        checkGenericErrorResponse(expect, response);
-        expect(response.statusCode).toEqual(400);
-        let errorPayload = JSON.parse(body);
-        expect(errorPayload.htsget).toBeDefined();
-        expect(errorPayload.htsget.error).toBe(ServerHttpError.INVALID_INPUT);
-        expect(response.statusMessage).toEqual(
-          'Invalid character in reference name chr@1');
-        done();
+    it(`redirection error, invalid characted in reference name for ${optionsList.method}`, function(done) {
+      let obj = JSON.parse(JSON.stringify(optionsList));
+      if (obj.method === "GET") {
+        obj.path = server_path + '?referenceName=chr@1&start=4&end=400';
+      } else {
+        obj.path = server_path;
+      }
+      let req = http.request ( obj, function(response) {
+        var body = '';
+        response.on('data', function(d) { body += d;});
+        response.on('end', function() {
+          checkGenericErrorResponse(expect, response);
+          expect(response.statusCode).toEqual(400);
+          let errorPayload = JSON.parse(body);
+          expect(errorPayload.htsget).toBeDefined();
+          expect(errorPayload.htsget.error).toBe(ServerHttpError.INVALID_INPUT);
+          expect(response.statusMessage).toEqual(
+            'Invalid character in reference name chr@1');
+          done();
+        });
       });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam", "accession":id, "regions":[{"referenceName":"chr@1", "start":4, "end":400}]}), () => {
```

Comment:
I was not expecting you to need to specify a `accession` here. Can you delete it and see if test still runs?